packages/block-editor/src/components/global-styles/use-global-styles-output.js

```diff
@@ -678,11 +678,11 @@ export const getNodesWithSettings = ( tree, blockSelectors ) => {
 	}
 
 	const pickPresets = ( treeToPickFrom ) => {
-		const presets = {};
+		let presets = {};
 		PRESET_METADATA.forEach( ( { path } ) => {
 			const value = get( treeToPickFrom, path, false );
 			if ( value !== false ) {
-				set( presets, path, value );
+				presets = setImmutably( presets, path, value );
```

Comment:
Pretty straightforward to trace - `presets` is already declared to be an object, and `path` is already an array of strings (see `PRESET_METADATA`  declaration.